app/Http/Controllers/ConnectionController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+use App\Connection;
+use Illuminate\Support\Facades\Auth;
+
+class ConnectionController extends Controller
+{
+    public function show($id)
+    {
+        $user = User::findOrFail($id);
+        $connections = Connection::getAvailableConnections($user);
+
+        return view('users.connections')
+            ->with('user', $user)
+            ->with('connections', $connections);
+    }
+
+    public function add($id)
+    {
+        $connection = new Connection();
+
+        $connection->owner_id = Auth::user()->id;
+        $connection->user_id = $id;
+
+
+        if(count(Connection::where('owner_id', $connection->owner_id)->where('user_id', $connection->user_id)->get()) > 0)
```

Comment:
This seems like an arcane pile of confusion.

Which do you think is more clear, what you have here or `Auth::user()->hasAlreadyConnectedWith($target_user)`?

Follow the 4 simple rules, listed in order of importance

1. Works ✅
2. Expresses intent 👎
3. No Duplication
4. Least lines of code

Reply:
Put the pile in user class 👍